File: bl-central-server/bl-merchandise-data-collector/src/main/java/org/jesperancinha/logistics/mcs/converter/MerchandiseLogConverter.java
package org.jesperancinha.logistics.mcs.converter;

import org.jesperancinha.logistics.jpa.model.Merchandise;
import org.jesperancinha.logistics.jpa.model.MerchandiseLog;

public class MerchandiseLogConverter {

    public static Merchandise toMerchandise(MerchandiseLog merchandiseLog) {
        return Merchandise.builder()
            .productCargo(merchandiseLog.getProductCargo())
            .supplier(merchandiseLog.getSupplier())
            .vendor(merchandiseLog.getVendor())
            .timestamp(merchandiseLog.getTimestamp())
            .transportPackage(merchandiseLog.getTransportPackage())
            .build();
    }
}

File: bl-central-server/bl-merchandise-data-collector/src/main/kotlin/org/jesperancinha/logistics/mcs/converter/MerchandiseLogConverter.kt
package org.jesperancinha.logistics.mcs.converter

import org.jesperancinha.logistics.jpa.dao.Merchandise
import org.jesperancinha.logistics.jpa.dao.MerchandiseLog

object MerchandiseLogConverter {
    fun toMerchandise(merchandiseLog: MerchandiseLog): Merchandise {
        return Merchandise(
            productCargo = merchandiseLog.productCargo,
            supplier = merchandiseLog.supplier,
            vendor = merchandiseLog.vendor,
            timestamp = merchandiseLog.timestamp,
            transportPackage = merchandiseLog.transportPackage,
            lat = merchandiseLog.lat,
            lon = merchandiseLog.lon
        )
    }
}